src/utils/FormattingUtils.ts

```diff
@@ -0,0 +1,165 @@
+ * @returns The shortened hexadecimal string.
+ * @example shortenHexString("0x772871a444c6e4e9903d8533a5a13101b74037158123e6709470f0afbf6e7d94") -> "0x7787...7d94"
+ */
+export function createShortHexString(hex: string): string {
+  return hex.substring(0, 5) + "..." + hex.substring(hex.length - 6, hex.length);
+}
+
+/**
+ * Generates a valid etherscan link for a given transaction hash or address.
+ * @param hex The transaction hash or address to link to.
+ * @param chainId The chainId to link to.
+ * @returns A formatted markdown etherscan link to the given transaction hash or address on the given chainId.
+ */
+function _createEtherscanLinkMarkdown(hex: string, chainId = 1): string | null {
+  if (hex.substring(0, 2) != "0x") {
+    return null;
+  }
+  const shortURLString = createShortHexString(hex);
+  // Transaction hash
+  if (hex.length == 66) {
+    return `<${createEtherscanLinkFromTx(chainId)}tx/${hex}|${shortURLString}>`;
+  }
+  // Account
+  else if (hex.length == 42) {
+    return `<${createEtherscanLinkFromTx(chainId)}address/${hex}|${shortURLString}>`;
+  }
+  return null;
+}
+
+/**
+ * Generates a list of etherscan links for a given list of transaction hashes or addresses.
+ * @param txHashesOrAddresses The list of transaction hashes or addresses to link to.
+ * @param chainId The chainId to link to.
+ * @returns A list of formatted markdown etherscan links to the given transaction hashes or addresses on the given chainId.
+ * @see etherscanLink
+ */
+export function etherscanLinks(txHashesOrAddresses: string[], chainId: number | string): string {
+  return txHashesOrAddresses.map((hash) => `${etherscanLink(hash, chainId)}\n`).join("");
+}
+
+/**
+ * Converts a utf8 string to a hex string.
+ * @param input The utf8 string to convert.
+ * @returns The hex string.
+ */
+export function utf8ToHex(input: string): string {
+  return ethers.utils.formatBytes32String(input);
+}
+
+/**
+ * Converts a hexadecimal string to a utf8 string.
+ * @param input The hexadecimal string to convert.
+ * @returns The utf8 string.
+ */
+export function hexToUtf8(input: string): string {
+  return ethers.utils.toUtf8String(input);
+}
+
+/**
+ * Converts a BigNumber to a 32-byte hexadecimal string.
+ *
+ * @param input - The BigNumber to convert.
+ * @returns The 32-byte hexadecimal string representation of the input.
+ */
+export function bnToHex(input: BigNumber): string {
+  return ethers.utils.hexZeroPad(ethers.utils.hexlify(toBN(input)), 32);
+}
+
+/**
+ * Converts a value from wei to a decimal value with the specified number of decimal places.
+ * @param weiVal - The value in wei to convert.
+ * @param decimals - The number of decimal places to include in the converted value.
+ * @returns The converted value as a string.
+ */
+export function convertFromWei(weiVal: string, decimals: number): string {
+  const formatFunction = createFormatFunction(2, 4, false, decimals);
+  return formatFunction(weiVal);
+}
```

Comment:
Aren't we conflating the PR by bundling all this in here?

FWIW I'm happy to see it out of `src/utils/common.ts` but the BN stuff doesn't really have much to do with block explorers. In general the concept of `FormattingUtils` is fairly vague to me. I think this fits better in a dedicated BigNumber file instead.

Reply:
Refactored